deploy/cr/tls/cr_minimal_with_tls.yaml

```diff
@@ -0,0 +1,15 @@
+apiVersion: infinispan.org/v1
+kind: Infinispan
+metadata:
+  name: example-infinispan
+spec:
+  replicas: 2
+  security:
+    endpointEncryption:
```

Comment:
This configuration makes it hard to choose one or the other option. What happens if someone puts both `certSecretName` and `certService`?

I think this would benefit from a `endpointEncryption.type` attribute that defines which type of encryption is chosen, and based on that we can validate that the right properties are set. It's the same we're doing with `spec.service` where we'll have either cache or datagrid...etc.

Reply:
Yaml comments explain the policy (with a typo actually). The choice is implicit: OS service if certService not null else certSecretName is used.
But I think it's OK to add the type field and uniform encryption to spec.service

Reply:
@galderz added a test for cluster formation with TLS

Reply:
@rigazilla having `type: service` commented out means community users have to manually edit the yaml instead of doing `oc apply -f github/raw/path/to/cr_minimal_with_tls.yaml`

what if we broke this into two files? 

`cr_minimal_service_encryption.yaml` - uses `type: service` 
`cr_minimal_custom_encryption.yaml` - uses `type: secret`